pkg/odp/lru_cache.go

```diff
@@ -0,0 +1,110 @@
+/****************************************************************************
+ * Copyright 2022, Optimizely, Inc. and contributors                        *
+ *                                                                          *
+ * Licensed under the Apache License, Version 2.0 (the "License");          *
+ * you may not use this file except in compliance with the License.         *
+ * You may obtain a copy of the License at                                  *
+ *                                                                          *
+ *    http://www.apache.org/licenses/LICENSE-2.0                            *
+ *                                                                          *
+ * Unless required by applicable law or agreed to in writing, software      *
+ * distributed under the License is distributed on an "AS IS" BASIS,        *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+ * See the License for the specific language governing permissions and      *
+ * limitations under the License.                                           *
+ ***************************************************************************/
+
+// Package odp //
+package odp
+
+import (
+	"container/list"
+	"sync"
+	"time"
+)
+
+// Cache is used for caching ODP segments
+type Cache interface {
+	Save(key string, value interface{})
+	Lookup(key string) interface{}
+	Reset()
+}
+
+type cacheElement struct {
+	data   interface{}
+	time   int64
+	keyPtr *list.Element
+}
+
+// LRUCache a Least Recently Used in-memory cache
+type LRUCache struct {
+	queue         *list.List
+	items         map[string]*cacheElement
+	maxSize       int
+	timeoutInSecs int64
+	lock          *sync.RWMutex
+}
+
+// NewLRUCache returns a new instance of Least Recently Used in-memory cache
+func NewLRUCache(size int, timeoutInSecs int64) LRUCache {
```

Comment:
To avoid converting `timeoutInSecs` to in64 every time `isValid` is called since `time.Now().Unix()` returns a `int64` value and calculations are only allowed between values of similar type's.